rcl/src/rcl/subscription.c

```diff
@@ -116,8 +119,12 @@ rcl_subscription_init(
   }
   subscription->impl->actual_qos.avoid_ros_namespace_conventions =
     options->qos.avoid_ros_namespace_conventions;
-  // options
-  subscription->impl->options = *options;
+  ret = rcl_subscription_options_copy(options, &subscription->impl->options);
+  if (RCL_RET_OK != ret) {
+    ret = RCL_RET_ERROR;
```

Comment:
why we need to overwrite the error code here?

Reply:
updated